### DISCUSSION TITLE
Proposal: Add tutorial pages for more easy first steps for newcomers.

### DISCUSSION THREAD

**Discussion by ybutski-altir** [keycloakify/keycloakify#193] [Ideas]
Hi! I am new in keycloak and even more new in keycloakify. I just searching in web for comfortable way how to customize Keycloak themes and found Keycloakify, where my first impression was "Wow! This looks cool!". And now, after reading all documentation on https://docs.keycloakify.dev/, viewing all examples links and looking on 2 starters (CSS and Components levels) I understand that I still have many questions how I can customize, what I can customuze, what I can use (Can I use Material UI or not and other React things?).
Documentation articles feels like many pieces of something common. And I feel like something common missed.

I found links to 2 demo projects and 1 real life project. And sure I will try to dig into them and understand all pieces. But I would suggest to add 2 tutorial pages (one for CSS and another for Component levels customization) with content like "So here we have starter project and we now know how to run and test theme locally. And now, here is design of our custom Login/Register page and let's try to customise default theme to achieve this result." And walk through all aspects - base customization, i18n, terms and cond. and etc. Or at least record video like this on youtube.

It just my proposal, I understand that you already made great effort to this!

**Comment by garronej** [comment]
Hi @ybutski-altir,  
Thanks for the feedback and the kind words :)  

> Can I use Material UI or not and other React things?  

Yes, of course, you can, we do. You can use whatever React library you'd like Keycloakify is agnostic to that.  

>  found links to 2 demo projects and 1 real life project.  

There is also this real-world example: https://github.com/InseeFrLab/onyxia-web/tree/main/src/ui/components/KcApp  
This is the login page of [this website](https://datalab.sspcloud.fr).  

> And sure I will try to dig into them and understand all pieces.  

I think I made sure that everything is covered in [keycloakify.dev](https://docs.keycloakify.dev/how-to-use).  
I provide the demo projects for peoples that prefers hacking around in working examples rather than reading the documentation (this is my case).  

Now that being said I do assume that readers of the documentation are already familiar with React and TypeScript.  

I do understand that video tutorials would make this tool acceiceible to more junior React devs and be an overall best learning experience for most people.  You are not the first to request videos tutorials.   

I don't do them for two reasons:    
- I am involved in maintaining [too many open source project](https://github.com/garronej) I don't have the bandwidth. I'm not used to producing video tutorials, this would require many, many takes and heavy editing to come up with something half decent. (Actually, I tried early on but [the result](https://github.com/InseeFrLab/keycloakify/releases/download/v0.0.1/keycloak-react-theming_quickstart_with_sound.mp4) was so lame that [I published the vide without sound](https://www.youtube.com/watch?v=xTz0Rj7i2v8)).  
- I want to be free to keep improving Keycloakify, It's already very time consuming to keep the documentation up to date. If I had to re-shoot a video for every major release, it would slow the development quite a bit.  

So that's it. I won't do it myself but I am hoping the community will eventually.  
I have an [hacktoberfest-accepted issue open about it](https://github.com/InseeFrLab/keycloakify/issues/188).  

Anyway, again, thanks for the nice feedback.  

Best,
